From ec82b217ba13589c22782ecc528ccc9c5a84d684 Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Mon, 22 Jun 2015 04:31:25 +0530
Subject: [PATCH] EM/TYPE-C: Fix to power on OTG during reboot

When the OTG is connected while rebooting, the peripheral device is
not powering on. This change will resolve the issue by sending
DRP event while booting based on the WAKE bit status

Change-Id: Ia6d25de056831472eb98cdc8e3b2366c1f744ef8
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-36254
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index e5faa4e39af9..a6ec07456fd6 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -1358,7 +1358,7 @@ static int fusb300_probe(struct i2c_client *client,
 	struct i2c_adapter *adapter = to_i2c_adapter(client->dev.parent);
 	struct fusb300_chip *chip;
 	int ret;
-	unsigned int val;
+	unsigned int val, stat;
 
 	if (!i2c_check_functionality(adapter, I2C_FUNC_SMBUS_WORD_DATA))
 		return -EIO;
@@ -1454,6 +1454,17 @@ static int fusb300_probe(struct i2c_client *client,
 
 	if (!chip->i_vbus) {
 		fusb300_wake_on_cc_change(chip);
+		regmap_read(chip->map, FUSB300_STAT0_REG, &stat);
+
+		if (stat &  FUSB300_STAT0_WAKE) {
+			if (chip->is_fusb300)
+				atomic_notifier_call_chain(&chip->phy.notifier,
+					TYPEC_EVENT_DRP, &chip->phy);
+			else
+				 atomic_notifier_call_chain(&chip->phy.notifier,
+					TYPEC_EVENT_DFP, &chip->phy);
+		}
+
 	} else
 		atomic_notifier_call_chain(&chip->phy.notifier,
 				TYPEC_EVENT_VBUS, &chip->phy);
-- 
1.9.1

